skrifa/src/outline/autohint/latin/blues.rs

```diff
@@ -436,6 +435,153 @@ fn compute_default_blues(font: &FontRef, coords: &[F2Dot14], style: &StyleClass)
     blues
 }
 
+/// Compute unscaled blue values for the CJK script set.
+///
+/// See <https://gitlab.freedesktop.org/freetype/freetype/-/blob/57617782464411201ce7bbc93b086c1b4d7d84a5/src/autofit/afcjk.c#L277>
+fn compute_cjk_blues(font: &FontRef, coords: &[F2Dot14], style: &StyleClass) -> [UnscaledBlues; 2] {
+    let mut blues = [UnscaledBlues::new(), UnscaledBlues::new()];
+    let mut outline_buf = UnscaledOutlineBuf::<MAX_INLINE_POINTS>::new();
+    let mut flats = [0; BLUE_STRING_MAX_LEN];
+    let mut fills = [0; BLUE_STRING_MAX_LEN];
+    let glyphs = font.outline_glyphs();
+    let charmap = font.charmap();
+    let units_per_em = font
+        .head()
+        .map(|head| head.units_per_em())
+        .unwrap_or_default() as i32;
+    // Walk over each of the blue character sets for our script.
+    for (blue_chars, blue_flags) in style.script.blues {
+        let is_horizontal = blue_flags & blue_flags::CJK_HORIZ != 0;
+        // Note: horizontal blue zones are disabled by default and have been
+        // for many years in FreeType:
+        // See <https://gitlab.freedesktop.org/freetype/freetype/-/blob/57617782464411201ce7bbc93b086c1b4d7d84a5/src/autofit/afcjk.c#L35>
+        // and <https://gitlab.freedesktop.org/freetype/freetype/-/commit/084abf0469d32a94b1c315bee10f621284694328>
+        if is_horizontal {
+            continue;
+        }
+        let is_right = blue_flags & blue_flags::CJK_RIGHT != 0;
+        let is_top = blue_flags & blue_flags::TOP != 0;
+        let blues = &mut blues[!is_horizontal as usize];
+        if blues.len() >= MAX_BLUES {
+            continue;
+        }
+        let mut n_flats = 0;
+        let mut n_fills = 0;
+        let mut is_fill = true;
+        for ch in *blue_chars {
+            if *ch == '|' {
+                is_fill = false;
+                continue;
+            }
+            // TODO: do some shaping
```

Comment:
Unless the shaping is coming soon I'd prefer an issue and a reference to that issue

Reply:
#1128